File: apps/backend/services/Business/Business.Presentation/Endpoints/Assets/UpdateAsset.Request.cs
namespace Business.Presentation.Endpoints.Assets;

/// <summary>JSON body for PUT /assets/{id}.</summary>
public record UpdateAssetRequest(
    string Name,
    Guid? CategoryId,
    string? Description,
    decimal? SalePrice,
    decimal? RentalPrice,
    int StockCount,
    string? ImgUrl);

File: apps/mobile/lib/core/features/inventory/models/requests/update_asset_request.dart
/// JSON body for PUT /assets/{id} — mirrors the Business API's UpdateAssetRequest.
class UpdateAssetRequest {
  final String name;
  final String? categoryId;
  final String? description;
  final double? salePrice;
  final double? rentalPrice;
  final int stockCount;
  final String? imgUrl;

  const UpdateAssetRequest({
    required this.name,
    required this.stockCount,
    this.categoryId,
    this.description,
    this.salePrice,
    this.rentalPrice,
    this.imgUrl,
  });

  Map<String, dynamic> toJson() => {
        'name': name,
        'categoryId': categoryId,
        'description': description,
        'salePrice': salePrice,
        'rentalPrice': rentalPrice,
        'stockCount': stockCount,
        'imgUrl': imgUrl,
      };
}
